### DISCUSSION TITLE
GPS Data wrong with T-echo

### DISCUSSION THREAD

**Discussion by dg3vz** [meshtastic/Discussions#30] [nRF52]
Hello,

I always get incorrect GPS data from my T-Echo.
The position given is always the same. But no fixed position is configured.
The correct position is shown on the display, but the wrong one is sent. It is also incorrect in the client on the iPhone.
But it worked once. I have already reset the T-echo to factory settings several times. (format and new firmware).
The firmware tested was 2.5.7 Beta and 2.5.8 Alpha. 
What else can I do?

Thanks

**Comment by RichieFrame** [comment]
What device role are you using?

**Reply by dg3vz** [reply]
Device role is client.
I also tried turning off GPS and setting a fixed position but it always gives the same wrong position.

**Comment by ikearat** [comment]
I went through a similar issue with my device. It would "default" to a position which was not what I set (and reset several times) and when provided actual GPS data, would "bounce" back and forth between the arbitrary location and the GPS data.
Another person suggested adding a second Channel, reassigning LongFast to that 2nd channel and then making Channel 0 (zero) a private channel.
This worked but I was unsatisfied and continued fooling with the Position and channels settings. After a little while and some jerking with the PSK field in the channel settings I "found" my device to be stable and accepting exactly the settings it was given.
My "final" settings are:
LoRa is set to the US (20)
Channel(s) are, Channel 0 is named "LongFast" and the PSK is "AQ==" (the AQ== thing took some digging, oddly people don't talk about the "default" PSK for Longfast.)
Channel 1 and the rest are unassigned.
Position settings are, Smart is "off", Fixed is "on" with my preferred location being the local city park parking lot.
The system is stable now and I can alter these settings as I feel like it with predictable results.
My "feeling" is that the factory configuration(s) for devices don't apply values in an order respecting dependencies in the firmware. After a cycle (or two) of stacking parameters manually, the bits of data the system needs are now in the correct formats and locations. As a support for this "feeling" my manual coordinates are now only XX.XXXXXX format where the nonsense positions that were being substituted were 16 decimal places on precision.
